exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/VarLenBulkPageReader.java

```diff
@@ -106,11 +107,13 @@ final void set(PageDataInfo pageInfoInput, boolean clear) {
     pageInfo.dictionaryValueReader = pageInfoInput.dictionaryValueReader;
     pageInfo.numPageValues = pageInfoInput.numPageValues;
     if (clear) {
-    buffer.clear();
-  }
+      buffer.clear();
+    }
   }
 
   final VarLenColumnBulkEntry getEntry(int valuesToRead) {
+    Preconditions.checkArgument(valuesToRead > 0, String.format("Number of values to read [%d] should be greater than zero", valuesToRead));
```

Comment:
Please apply the same to all other places where there is a check for argument or state to be greater than zero.

Reply:
It may also be better to use Preconditions template instead of String template (sorry for the wrong example in my prior comment).